owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/http/methods/webdav/DavMethod.kt

```diff
@@ -27,45 +27,39 @@ import at.bitfire.dav4jvm.Dav4jvm.log
 import at.bitfire.dav4jvm.DavOCResource
 import at.bitfire.dav4jvm.exception.HttpException
 import at.bitfire.dav4jvm.exception.RedirectException
-import com.owncloud.android.lib.common.http.HttpClient
 import com.owncloud.android.lib.common.http.HttpConstants
 import com.owncloud.android.lib.common.http.methods.HttpBaseMethod
-import okhttp3.HttpUrl
-import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.OkHttpClient
 import okhttp3.Protocol
 import okhttp3.Response
 import okhttp3.ResponseBody.Companion.toResponseBody
-import java.net.MalformedURLException
 import java.net.URL
-import java.util.concurrent.TimeUnit
 
 /**
  * Wrapper to perform WebDAV (dav4android) calls
  *
  * @author David González Verdugo
  */
-abstract class DavMethod protected constructor(httpClient: HttpClient, url: URL) : HttpBaseMethod(httpClient, url) {
-    protected var davResource: DavOCResource
-
+abstract class DavMethod protected constructor(url: URL) : HttpBaseMethod(url) {
     override lateinit var response: Response
-
-    init {
-        val httpUrl = url.toHttpUrlOrNull() ?: throw MalformedURLException()
-        davResource = DavOCResource(
-            okHttpClient,
-            httpUrl,
-            log
-        )
-    }
+    private var davResource: DavOCResource? = null
 
     override fun abort() {
-        davResource.cancelCall()
+        davResource?.cancelCall()
     }
 
+    protected abstract fun onDavExecute(davResource: DavOCResource): Int
+
     @Throws(Exception::class)
-    override fun execute(): Int {
+    override fun onExecute(okHttpClient: OkHttpClient): Int {
         return try {
-            onExecute()
+             davResource = DavOCResource(
+                okHttpClient.newBuilder().followRedirects(false).build(),
```

Comment:
Why do we set `followRedirects(false)` here? And why follow redirects is true by default if we set it to false every time we perform a davMethod execution? Just curious 🤔

Reply:
dav methods don't allow to have redirects enabled. dav4android will throw if one of its methods get called with a client that hass redirects enabled. therefore this is a last safeguard to disable redirects. Lets for now please keep it like this. ConnectionValidater does change the redirect foo all together so this will not matter much.